rust-client/src/main.rs

```diff
@@ -99,26 +100,14 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let local_peer_id = PeerId::from(local_key.public());
     info!("Local peer id: {:?}", local_peer_id);
 
-    let mut protocols = None;
-
     for _ in 0..opt.rounds.unwrap_or(usize::MAX) {
         let mut swarm = init_swarm(local_key.clone()).await?;
-        if protocols.is_none() {
-            let supported_protocols = swarm
-                .behaviour_mut()
-                .new_handler()
-                .inbound_protocol()
-                .protocol_info()
-                .into_iter()
-                .map(|info| String::from_utf8(info.protocol_name().to_vec()))
-                .collect::<Result<Vec<String>, _>>()?;
-            let _ = protocols.insert(supported_protocols);
-        }
 
         let request = tonic::Request::new(grpc::RegisterRequest {
             client_id: local_peer_id.to_bytes(),
             agent_version: agent_version(),
-            protocols: protocols.clone().unwrap(),
+            // TODO: Send the right set of protocols.
```

Comment:
Interesting, I guess we don't expose this at the moment. Perhaps this should be written as a `NetworkBehaviour`?

Reply:
Possible. I suggest to delay it. I don't think the `protocols` field is needed as of today.